### ISSUE TITLE
change default LookupIpStrategy from `Ipv4AndIpv6` to `Ipv4thenIpv6`?

### ISSUE THREAD

**Issue by cssivision** [bluejekyll/trust-dns#300]
ipv6 is not well supported now, `Ipv4AndIpv6` Query for A and AAAA in parallel, the situation is that if the website you visited support ipv6, but your network do not support ipv6, you will get a `Network Unreachable` error.

**Comment by bluejekyll** [comment]
Can you describe your toolchain as your using it? That is, is this something that you could override locally? There are a few options for that.

The reason I picked `IPv4AndIPv6` as the default was to make sure that there wasn't a hit in lookup speed for `IPv4ThenIpv6`. I'm open to this, but would love additional input from other users.

**Comment by cssivision** [comment]
i use default config by function `from_system_conf`,  and my network only support ipv4, when i query dns for `www.google.com` sometimes it return ipv6 address, and my application failed.  

the `options` filed of  `ResolverFuture` struct is not `pub`, and there is no way for me to change this `ip_strategy` if i want to use system default config.

**Comment by cssivision** [comment]
maybe we can make this `options` field `pub`?

**Comment by bluejekyll** [comment]
I'd prefer not making that field `pub` I honestly don't know what would happen if you change the config after initialization, and don't know if I want to be restricted by that if we do allow it.

Based on your use case, I wonder if we should change this default, possible by platform. What system are you on? I do this this matches the way glibc performs lookups on linux, from the resolv.conf man pages:

```text
... By default, glibc
performs IPv4 and IPv6 lookups in parallel since
version 2.9. ...
```

----
Example to override system conf:

```rust
let (mut config, mut options) = system_conf::read_system_conf()?;
options.... = ...;
config.... = ...;
let resolver = ResolverFuture::new(config, options, reactor))?;
```

**Comment by cssivision** [comment]
according to `RFC 6724` it seems that we should respect the order provided by getaddrinfo(3). It is already configured to use IPv6 first if both sides have native IPv6, but use IPv4 first if both sides are tunnelling. See RFC 6724 (the new version of RFC 3484), which is implemented in most libc.

**Comment by bluejekyll** [comment]
https://tools.ietf.org/html/rfc6724 has a lot of details that are still unimplemented by the Resolver...

Maybe we should open a separate issue, or rename this one, to instead be an enhancement to properly implement that RFC? It will take some work, as I don't think the Resolver currently supports options around a lot of the features being dictated in that yet.

**Comment by cssivision** [comment]
currently, resolver picks IPv4/IPv6 at random, i think we should have a determined operations, otherwise the users can't make up its mind.

now, i write code like this to fix the problems.
```rust
if let Some(addr) = res.iter().filter(|addr| addr.is_ipv4()).next() {
            future::ok(addr)
        } else {
            future::err(io::Error::new(
                io::ErrorKind::Other,
                "resolve fail".to_string(),
            ))
}
```

in my opinion, before this problem is solved, make `Ipv4thenIpv6` as a default `LookupIpStrategy` is a good solution.

**Comment by bluejekyll** [comment]
fair enough. Would you want to submit a PR?

**Comment by cssivision** [comment]
yes

**Comment by bluejekyll** [comment]
Fixed in #301
